pkg/controller/replica.go

```diff
@@ -133,7 +134,9 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
 	emptyCtx := context.TODO()
 	deletionCtx, cancelFunc := context.WithCancel(emptyCtx)
-	if _, ok := r.cleanUpMap.LoadOrStore(volumeName, cancelFunc); ok {
+	initChannel := time.After(r.timeUntilGarbageCollection)
+
+	if _, ok := r.cleanUpMap.LoadOrStore(volumeName, cleanUpEntry{ctx: deletionCtx, cancelFunc: cancelFunc, initChannel: initChannel}); ok {
```

Comment:
Consider using a simple `Load()` first, and only create the `cleanUpEntry` and use `LoadOrStore` if the initial `Load()` returns `!ok`. This way we avoid creating and immediately tearing down a complex struct with system resources unconditionally.